### PR TITLE
docs(internal): Add template for releasing VRL

### PR DESCRIPTION
**Pull request by pront**
## Summary

Documentation for the release process.

## Change Type

- [ ] Bug fix
- [ ] New feature
- [x] Non-functional (chore, refactoring, docs)
- [ ] Performance

## Is this a breaking change?

- [ ] Yes
- [x] No

## Does this PR include user facing changes?

- [ ] Yes. Please add a changelog fragment based on
  our [guidelines](https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md).
- [x] No. A maintainer will apply the "no-changelog" label to this PR.

## Checklist

- [x] Our [CONTRIBUTING.md](https://github.com/vectordotdev/vrl/blob/main/CONTRIBUTING.md) is a good starting place.
- [x] If this PR introduces changes to [LICENSE-3rdparty.csv](https://github.com/vectordotdev/vrl/blob/main/LICENSE-3rdparty.csv), please
  run `dd-rust-license-tool write` and commit the changes. More details [here](https://crates.io/crates/dd-rust-license-tool).
- [x] For new VRL functions, please also create a sibling PR in Vector to document the new function.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/minor-release.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+name: VRL minor release
+about: Use this template for a new minor release.
+title: "VRL [version] release"
+labels: "domain: releasing"
+---
+
+- [ ] Create release preparation PR
+  - [ ] Bump [Cargo.toml](https://github.com/vectordotdev/vrl/blob/main/Cargo.toml#L3) version and commit the change.
```
Comment (jszwedko) [inline .github/ISSUE_TEMPLATE/minor-release.md:9]: I'd suggest adding to run `cargo check` to update `Cargo.lock` too.

File: .github/ISSUE_TEMPLATE/minor-release.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  - [ ] Run the [generate_release_changelog.sh](https://github.com/vectordotdev/vrl/blob/main/scripts/generate_release_changelog.sh) script
```
Comment (jszwedko) [inline .github/ISSUE_TEMPLATE/minor-release.md:10]:
```suggestion
  - [ ] Run the [./scripts/generate_release_changelog.sh](https://github.com/vectordotdev/vrl/blob/main/scripts/generate_release_changelog.sh) script
```

File: .github/ISSUE_TEMPLATE/minor-release.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    and commit the changes.
+- [ ] After the above PR is merged, run the [publish.py](https://github.com/vectordotdev/vrl/blob/main/scripts/publish.py) script.
```
Comment (jszwedko) [inline .github/ISSUE_TEMPLATE/minor-release.md:12]:
```suggestion
- [ ] After the above PR is merged, run the [scripts/publish.py](https://github.com/vectordotdev/vrl/blob/main/scripts/publish.py) script.
```

**Review by jszwedko — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/minor-release.md (hunk 1 above)
Comment (pront) [reply .github/ISSUE_TEMPLATE/minor-release.md:9]: Good point, added. (My IDE does it automatically so I always skipped this)

**Pull request merged**